### ISSUE TITLE
Nested Fullscreen Exit breaks VJS Player

### ISSUE THREAD

**Issue by richardbushell** [videojs/video.js#5828]
Hi

I understand there is a review forthcoming of Fullscreen handling in VJS, but I would really appreciate for now any thoughts/solutions for why VJS is not returning from a nested fullscreen state correctly.

I have provided two reduced test cases below:-
VJS Video Player:  https://living.video/vjsreducednested.html
Standard Video Element (non-VJS):  https://living.video/videonestedfullscreen.html

Steps to reproduce:  Click Toggle Fullscreen Text below video (enters Document Fullscreen), Start Playing Video, Click Video Fullscreen enter, Click Video Fullscreen exit

In the first case (with VJS) the video disappears, in the second case (without VJS) the video element continues playing normally, with the document remaining in fullscreen as expected.

We have a live implementation of a Single-Page Web App (a website designed to view fullscreen throughout usage), but we cannot find a solution to this.

Any help would be HUGELY appreciated so that we may finalise usage.

**Comment by richardbushell** [comment]
p.s. problem may also be related to:   https://github.com/videojs/video.js/issues/5814

**Comment by richardbushell** [comment]
OK, possibly found a BUG in the code within: documentFullscreenChange_ function :-
https://docs.videojs.com/player.js.html#line1965
```
	this.isFullscreen(document[fsApi.fullscreenElement]);
```
I believe this should be:-
```
		this.isFullscreen(document[fsApi.fullscreenElement] === this.el());
```
Totally untested, but Fullscreen API uses a LIFO stack. Here it looks like you were only confirming it's not null, but you need to confirm it was the player element and not the HTML Document (or another element). I haven't looked yet whether this is replicated elsewhere in the code too.

Let me know your thoughts, if you agree I can send a PR.

**Comment by richardbushell** [comment]
Hi, I just tested this and it solves the Bug, see example here:-
https://living.video/vjsreducednestedfix.html

So, I have issued this as a Pull Request here:
https://github.com/videojs/video.js/pull/5830

**Comment by richardbushell** [comment]
Closed via 3fbc4f5
